### PR TITLE
impl GetAllCoinsInfoService

### PR DESCRIPTION
**Pull request by qct**
Implement of get all coins information: 
GET /sapi/v1/capital/config/getall (HMAC SHA256)

response:
```
[
    {
        "coin": "BTC",
        "depositAllEnable": true,
        "free": "0.08074558",
        "freeze": "0.00000000",
        "ipoable": "0.00000000",
        "ipoing": "0.00000000",
        "isLegalMoney": false,
        "locked": "0.00000000",
        "name": "Bitcoin",
        "networkList": [
            {
                "addressRegex": "^(bnb1)[0-9a-z]{38}$",
                "coin": "BTC",
                "depositDesc": "Wallet Maintenance, Deposit Suspended", // 仅在充值关闭时返回
                "depositEnable": false,
                "isDefault": false,        
                "memoRegex": "^[0-9A-Za-z\\-_]{1,120}$",
                "minConfirm": 1,  // 上账所需的最小确认数
                "name": "BEP2",
                "network": "BNB",            
                "resetAddressStatus": false,
                "specialTips": "Both a MEMO and an Address are required to successfully deposit your BEP2-BTCB tokens to Binance.",
                "unLockConfirm": 0,  // 解锁需要的确认数 
                "withdrawDesc": "Wallet Maintenance, Withdrawal Suspended", // 仅在提现关闭时返回
                "withdrawEnable": false,
                "withdrawFee": "0.00000220",
                "withdrawIntegerMultiple": "0.00000001",
                "withdrawMax": "9999999999.99999999",
                "withdrawMin": "0.00000440",
                "sameAddress": true  // 是否需要memo
            },
            {
                "addressRegex": "^[13][a-km-zA-HJ-NP-Z1-9]{25,34}$|^(bc1)[0-9A-Za-z]{39,59}$",
                "coin": "BTC",
                "depositEnable": true,
                "isDefault": true,
                "memoRegex": "",
                "minConfirm": 1,  // 上账所需的最小确认数
                "name": "BTC",
                "network": "BTC",
                "resetAddressStatus": false,
                "specialTips": "",
                "unLockConfirm": 0,  // 解锁需要的确认数
                "withdrawEnable": true,
                "withdrawFee": "0.00050000",
                "withdrawIntegerMultiple": "0.00000001",
                "withdrawMax": "750",
                "withdrawMin": "0.00100000",
                "sameAddress": false
            }
        ],
        "storage": "0.00000000",
        "trading": true,
        "withdrawAllEnable": true,
        "withdrawing": "0.00000000"
    }
]
```

### CODE REVIEW

File: v2/asset_detail_service.go [hunk 1]
```diff
@@ -42,6 +42,28 @@ func (s *GetAssetDetailService) Do(ctx context.Context) (res map[string]AssetDet
 	return res, nil
 }
 
+type GetAllCoinsInfoService struct {
+	c     *Client
+	asset *string
+}
+
+func (s *GetAllCoinsInfoService) Do(ctx context.Context) (res []CoinInfo, err error) {
```
Comment (adshao) [inline v2/asset_detail_service.go:50]:
```suggestion
func (s *GetAllCoinsInfoService) Do(ctx context.Context) (res []*CoinInfo, err error) {
```
Use pointer here to keep consistent with other services.

**Review by adshao — CHANGES_REQUESTED** [review]

File: v2/asset_detail_service.go [hunk 2]
```diff
@@ -42,6 +42,28 @@ func (s *GetAssetDetailService) Do(ctx context.Context) (res map[string]AssetDet
...
+	asset *string
+}
+
+func (s *GetAllCoinsInfoService) Do(ctx context.Context) (res []*CoinInfo, err error) {
+	r := &request{
+		method:   http.MethodGet,
+		endpoint: "/sapi/v1/capital/config/getall",
+		secType:  secTypeSigned,
+	}
+	data, err := s.c.callAPI(ctx, r)
```
Comment (adshao) [inline v2/asset_detail_service.go:56]:
```
        data, err := s.c.callAPI(ctx, r, opts...)
	if err != nil {
		return []*CoinInfo{}, err
	}
	res = make([]*CoinInfo, 0)
	err = json.Unmarshal(data, &res)
	if err != nil {
		return []*CoinInfo{}, err
	}
	return res, nil
```

**Review by adshao — CHANGES_REQUESTED** [review]

**Comment by adshao** [comment]
Hi @qct , could you please run this test check on https://github.com/adshao/go-binance/blob/master/.travis.yml, there are gofmt/golint issues on your code.

**Comment by MJ23M** [comment]
Hi, @adshao can you merge this pull request into master?

**Comment by qct** [comment]
@adshao sorry for the late, seems someone has done the job, close this pr.

**Comment by byroncoetsee** [comment]
Is there documentation for this anywhere?

**Pull request closed without merging**